crates/aranya-metrics/src/backend/mod.rs

```diff
@@ -0,0 +1,125 @@
+//! This module contains all the backend code needed to export metrics to a remote server, as well
+//! as various other configuration parameters that handle things such as logging to the console.
+//!
+//! The currently supported backends are:
+//! * [Prometheus], using either [Prometheus](https://prometheus.io/) scraping, or a `pushgateway`.
+//! * [Datadog] using [DogStatsD], provided as a service as part of a Datadog Agent.
+//! * [TCP Server], a lightweight TCP server for more customized metrics collection and processing, sent using [`protobuf`].
+//!
+//! [Prometheus]: https://docs.rs/metrics-exporter-prometheus/
+//! [DataDog]: https://docs.rs/metrics-exporter-dogstatsd/
+//! [DogStatsD]: https://docs.datadoghq.com/developers/dogstatsd/
+//! [TCP Server]: https://docs.rs/metrics-exporter-tcp/
+//! [`protobuf`]: https://protobuf.dev/
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use anyhow::Result;
+
+#[cfg(feature = "datadog")]
+pub mod datadog;
+#[cfg(feature = "prometheus")]
+pub mod prometheus;
+#[cfg(feature = "tcp_server")]
+pub mod tcp_server;
```

Comment:
nit: I think these are all running a server. `tcp` is a backend just like `datadog` and `prometheus` so there's no need to specify that it's a server here.
```suggestion
#[cfg(feature = "tcp")]
pub mod tcp;
```